Dockerfile-Dev

```diff
@@ -1,36 +1,49 @@
-# ----- Base Java - Check Dependencies ----
-FROM maven:3.6.1-jdk-8 AS base
-WORKDIR /app
-ADD pom.xml /app
-
-#
-# ----Build App with Dependencies ----
-FROM base AS dependencies
-ADD . /app
-# Running mvn clean package will run all the tests as well as package the app up in the jar form
-RUN mvn clean package -DskipTests
-
-#
-# ---- Release App ----
-FROM  openjdk:8-jre-alpine AS release
-WORKDIR /app
-
-### install azure cli
-RUN \
-  apk add --no-cache curl tar openssl sudo bash jq python3 && \
-  apk add --virtual=build gcc libffi-dev musl-dev openssl-dev make python3-dev && \
-  pip3 --no-cache-dir install --upgrade pip && \
-  pip3 --no-cache-dir install azure-cli
-
-# Create a user
-RUN addgroup -g 4120 -S helium && \
-    adduser -u 4120 -S helium -G helium && \
-    mkdir -p /home/helium/.azure && \
-    chown -R helium:helium /home/helium
+# ---- build the helium base image ----
+FROM  ubuntu:latest AS helium-dev-base
+
+# set noninteractive installation so install works unattended and does not ask any interactive
+# questions.
+ARG DEBIAN_FRONTEND=noninteractive
+
+### install the ubuntu dev tooling and core utilities needed later
+RUN apt-get update && \
+    apt-get install -y sudo apt-utils apt-transport-https ca-certificates curl gnupg software-properties-common libssl-dev git wget nano lsb-release jq && \
+    apt-get upgrade -y
+
+# Install Azure Cli
+RUN curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.asc.gpg && \
+    CLI_REPO=$(lsb_release -cs) && \
+    echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ bionic main" > /etc/apt/sources.list.d/azure-cli.list && \
+    apt-get update && \
+    apt-get install -y openjdk-8-jdk && \
+    apt-get install -y maven && \
+    apt-get install -y azure-cli
+
+# Add the helium user with the default
+RUN useradd -u 1000 -U -m -s /bin/bash helium && \
+    usermod -aG sudo helium && \
+    gpasswd -a helium sudo &&  \
+    echo "helium ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/90-helium-users && \
+    mkdir -p /home/helium/.azure
+
+### build the final image with all the packages and put the code in there.
+FROM helium-dev-base
+
+RUN apt-get update && \
+    apt-get upgrade -y
```

Comment:
the update and upgrade area already in the base. Per our conversation,  it was our understanding that we wanted this in the second stage as well so that even if there are small drifts in the packages from the base, we would get the latest. 

See "Invoke apt update to update the packages from the base image" from the Design Review Considerations Section - https://github.com/retaildevcrews/helium-java/issues/220

Reply:
Line 9-11 before the apt-get I had to use sudo.
Line 14-20 I had to be root.
Line 23 - UID 1000 is not unique - a common problem.  You can get the highest UID and add 1 to it with this command:
export nextuid=$(cat /etc/passwd |awk -F: '{print $3 + 1}'|sort -n |tail -n 2|grep -v 6553|awk '{print $1 +1}')

Line 23 could then use the $nextuid variable

Reply:
The dockerfile runs as root until you change the user using the USER directive. This happens on line 38. All the system level commands take place before that so this is not necessary.  Also, adding the helium user is not happening on the host system, rather it is happening in the docker image being built where are there no other users and thus UID 1000 is available. The requirement is on the user building docker to use **their** UID from the host system and use that as the helium user ID in the dockerfile-dev so they match and after mounting, the docker user has write access to the .azure folder.

Reply:
Ok - thank you